### ISSUE TITLE
Concurrent map read/write panic

### ISSUE THREAD

**Issue by Botono** [nerdswords/yet-another-cloudwatch-exporter#273]
My instance of YACE crashed with this panic message: `fatal error: concurrent map read and map write`. The entire stack trace is available here: https://pastebin.com/j0PPcg2Z

I believe the relevant section is this:

```
main.recordLabelsForMetric(0xc000e8fcc0, 0x20, 0xc001478600)
	/opt/aws_cloudwatch.go:599 +0x65 fp=0xc0002d57d0 sp=0xc0002d56c8 pc=0xd8bc95
main.migrateCloudwatchToPrometheus(0xc0000ce000, 0x13e, 0x200, 0x6, 0xc00020ac80, 0x4)
	/opt/aws_cloudwatch.go:720 +0x3f3 fp=0xc0002d5968 sp=0xc0002d57d0 pc=0xd8d023
main.updateMetrics(0xc0000674a0)
	/opt/main.go:86 +0xa9 fp=0xc0002d5ac8 sp=0xc0002d5968 pc=0xd93009
main.main.func3(0x13878c0, 0xc0004dcfc0, 0xc000918600)
	/opt/main.go:148 +0x168 fp=0xc0002d5b60 sp=0xc0002d5ac8 pc=0xd97bc8
```
It looks like the issue is reading the global variable `labelMap` in `recordLabelsForMetric()`: https://github.com/ivx/yet-another-cloudwatch-exporter/blob/master/aws_cloudwatch.go#L599

**Comment by Botono** [comment]
Some further context: I run YACE with `--decoupled-scraping=false`

**Comment by goya** [comment]
yikes, i just cleaned up some code in that area which didnt work correctly

i might delve back in and see if it can be simplified as it's quite busy there.

**Comment by thomaspeitz** [comment]
Still a problem?

**Comment by cristiangreco** [comment]
This has been fixed with https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/555, released in 0.35.0.
